src/WebJobs.Script/Workers/Rpc/Configuration/RpcWorkerConfigFactory.cs

```diff
@@ -159,6 +168,59 @@ internal void AddProvider(string workerDir)
             }
         }
 
+        private List<WorkerDescriptionProfile> ReadWorkerDescriptionFromProfiles(JObject workerConfig)
+        {
+            List<WorkerDescriptionProfile> descriptionProfiles = new List<WorkerDescriptionProfile>();
+            JArray profiles = workerConfig.GetValue(RpcWorkerConstants.WorkerDescriptionProfiles) as JArray;
+            if (profiles != null)
+            {
+                try
+                {
+                    //descriptionProfiles = profiles.ToObject<List<WorkerDescriptionProfile>>();
+                    foreach (JObject item in profiles)
+                    {
+                        descriptionProfiles.Add(JsonConvert.DeserializeObject<WorkerDescriptionProfile>(item.ToString(Formatting.None), new WorkerDescriptionProfileConverter(_logger, _systemRuntimeInformation, _environment)));
```

Comment:
Do we use a custom JsonConverter `WorkerDescriptionProfileConverter` or reply on the default converter. This is the only place I could think of using the `IWorkerProfileConditionFactory`. Is there a better way to do this?

Reply:
Here we'd use the factory directly, after deserializing the profile and getting the type and dictionary of properties.